monkey/monkey_island/cc/setup/mongo/mongo_setup.py

```diff
@@ -43,15 +44,19 @@ def register_mongo_shutdown_callback(mongo_db_process: MongoDbProcess):
     atexit.register(mongo_db_process.stop)
 
 
-def connect_to_mongodb():
-    _wait_for_mongo_db_server(MONGO_URL)
+def connect_to_mongodb(mongo_db_process: MongoDbProcess):
+    _wait_for_mongo_db_server(MONGO_URL, mongo_db_process)
     _assert_mongo_db_version(MONGO_URL)
     mongo_connector.connect_dal_to_mongodb()
 
 
-def _wait_for_mongo_db_server(mongo_url):
+def _wait_for_mongo_db_server(mongo_url, mongo_db_process):
     while not is_db_server_up(mongo_url):
         logger.info("Waiting for MongoDB server on {0}".format(mongo_url))
+
+        if mongo_db_process is not None and not mongo_db_process.is_running():
+            raise MongoDbProcessException
+
```

Comment:
I like the idea of having this raise a timeout exception. And checking if the process is running in `server_setup.py`